tests/python/test_print.py

```diff
@@ -67,6 +67,72 @@ def func(k: ti.f32):
     ti.sync()
 
 
+@test_utils.test(exclude=[ti.dx11, vk_on_mac, ti.amdgpu], debug=True)
+def test_print_matrix_fstring():
+    x = ti.Matrix.field(2, 3, dtype=ti.f32, shape=())
+    y = ti.Vector.field(3, dtype=ti.f32, shape=3)
+
+    @ti.kernel
+    def func(k: ti.f32):
+        x[None][0, 0] = -1.0
+        y[2] += 1.0
+        print(f'hello {x[None]} world!')
+        print(f'{y[2] * k} {x[None] / k} {y[2]}')
+
+    func(233.3)
+    ti.sync()
+
+
+@test_utils.test(exclude=[ti.dx11, vk_on_mac, ti.amdgpu], debug=True)
+def test_print_matrix_fstring_with_spec():
+    x = ti.Matrix.field(2, 3, dtype=ti.f32, shape=())
+    y = ti.Vector.field(3, dtype=ti.f32, shape=3)
+    z = ti.Matrix.field(2, 3, dtype=ti.i32, shape=())
+
+    @ti.kernel
+    def func(k: ti.f32):
+        x[None][0, 0] = -1.0
+        y[2] += 1.0
+        print(f'hello {x[None]:.2f} world!')
+        print(f'{(y[2] * k):.3f} {(x[None] / k):e} {y[2]:.2}')
+        print(f'hello {z[None]:+d} world!')
+
+    func(233.3)
```

Comment:
Is there a way to capture the print message from stdout and verify that it's expected?

Reply:
Now stdout is captured by `capfd`, and we can check whether the output is desired~

Reply:
BTW, it seems that Python unit test for printing runs on **all** architectures except`dx11`, `vulkan_on_mac` and `amdgpu`. I corrected this by setting the desired architecture to  `[cpu, cuda, vulkan]`.

Reply:
yep, printing's only supported on cpu, cuda and vulkan so far. Thanks!